Unified/closor.py

```diff
@@ -317,6 +319,34 @@ def rank( wfn ):
 %s 
 This is an automated message.
 """%( bname, 
+      bname,
+      issues)
+            to = ['hn-cms-relval@cern.ch']
+            sendEmail(subject, text, destination=to )
+            ## just announced ; take it out now.
+            #BI.pop( bname )
+            subject = "Release Validation low statistics"
+            issues = ""
+            if batch_extreme_warnings[ bname ]:
+                issues="The following datasets have outstanding completion (<50%%) issues:\n\n"
+                issues+="\n".join( sorted( batch_extreme_warnings[ bname ] ))
+                issues+="\n\n"
+            text = """
+Dear all,
+
+a batch of release validation workflows has finished.
+
+Batch ID:
+
+%s
+
+Detail of the workflows
+
+https://dmytro.web.cern.ch/dmytro/cmsprodmon/requests.php?campaign=%s
+
+%s 
+This is an automated message.
+"""%( bname,
```

Comment:
The second `sendEmail` is outside of `if batch_extreme_warnings[ bname ]:` condition. You will send email twice every time regardless of final statistics.

To ensure code readability, I would suggest to reformat the main body text, something like:
```
    text = ""
    text += "Dear all,\n\n" # Double linebreak since HN format usually will truncate one.
    text += "A batch of ..." 
```

And then depends on the condition (`batch_warnings` or `batch_extreme_warnings`), you will add the proper issue and change the subject accordingly, and send email only *once*.

Reply:
@thongonary I am changing the subject line. For the email thing, I thought this has to be a second email in case it is < 50 else only one email. do you want to send only one email if the stats<50 ?